### PR TITLE
Use Context object consistently across code base

### PR DESCRIPTION
**Pull request by mkevinosullivan**
### WHY are these changes introduced?

`Context` provides file system, output, environment variable, system call, et. al. capabilities that are also being called directly in other parts of the code base.  Goal of this PR is to consistently use the capabilities of `Context` through the `Context` object.

### WHAT is this pull request doing?

- replacing direct calls:
  - to `FileUtils.rm()` with `ShopifyCli::Context.rm()`
  - to `File.write()` with `ShopifyCli::Context.write()`
  - to `FileUtils.mkdir_p()` with `ShopifyCli::Context.mkdir_p()`
  - to `CLI::UI.fmt()` with one of `#print_task()`, `#puts()`, `#done()` or `#error()` from `ShopifyCli::Context`
  - to `CLI::Kit::System.system` with `ShopifyCli::Context.system`
  - to `CLI::Kit::System.capture2` with `ShopifyCli::Context.capture2`
- refactoring code (e.g., use of `Helpers`) to use `Context` instead where possible.

### CODE REVIEW

File: lib/shopify-cli/commands/deploy/heroku.rb [hunk 1]
```diff
@@ -143,30 +141,30 @@ def heroku_command
...
-          raise(ShopifyCli::Abort, '{{x}} Heroku app could not be created') unless status.success?
+          @ctx.error('Heroku app could not be created') unless status.success?
           @ctx.puts(output)
 
           new_remote = output.split("\n").last.split("|").last.strip
           result = @ctx.system('git', 'remote', 'add', 'heroku', new_remote)
-          msg = "{{x}} Heroku app created, but couldn’t be set as a git remote"
-          raise(ShopifyCli::Abort, msg) unless result.success?
+          msg = "Heroku app created, but couldn’t be set as a git remote"
+          @ctx.error(msg) unless result.success?
```
Comment (ayronshopify) [inline lib/shopify-cli/commands/deploy/heroku.rb]: What was previously printed in `{{x}}`? Will that still be printed in the new `@ctx.error` flow?

File: lib/project_types/node/commands/create.rb [hunk 2]
```diff
@@ -66,18 +66,18 @@ def self.help
 
       def check_node
         version, stat = @ctx.capture2e(%w(node -v))
-        @ctx.puts("{{v}} node #{version}")
-        raise(ShopifyCli::Abort, NODE_REQUIRED_NOTICE) unless stat.success?
+        @ctx.done("node #{version}")
+        @ctx.error(NODE_REQUIRED_NOTICE) unless stat.success?
```
[the remark this answers is not in the document]
Comment (tanema) [reply lib/project_types/node/commands/create.rb]: would it be better if the name was abort, or raise?

File: lib/shopify-cli/commands/deploy/heroku.rb (hunk 1 above)
Comment (mkevinosullivan) [reply lib/shopify-cli/commands/deploy/heroku.rb]: `@ctx.error() ` prefixes `{{x}} ` to the message parameter passed to it, hence it's removed in any message that is now passed to `#error()`

File: lib/project_types/node/commands/create.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (mkevinosullivan) [reply lib/project_types/node/commands/create.rb]: Which would you consider better, `#raise` or `#abort`?  I'm somewhat indifferent either way, but `#abort()` might be a smidge better, as it indicates that an `Abort` will be `raise`d, vs. `raise()` which indicates that _something_ is `raise`d.

[the remark this answers is not in the document]
Comment (tanema) [reply lib/project_types/node/commands/create.rb]: yeah I would agree with you @mkevinosullivan abort sounds a but better

**Pull request merged**